web/impact/impact/tests/test_judging_round_list_view.py

```diff
@@ -100,3 +110,41 @@ def test_get_round_type_cannot_be_bogus(self):
             response = self.client.get(self.url + "?round_type=bogus")
             assert response.status_code == 401
             assert response.data == [INVALID_ROUND_TYPE_ERROR.format("bogus")]
+
+    def test_clearance_enforced(self):
+        the_round = JudgingRoundFactory.create(
+            round_type=ONLINE_JUDGING_ROUND_TYPE)
+        url = self.url + "?round_type={}".format(ONLINE_JUDGING_ROUND_TYPE)
+        user = self.basic_user()
+        # This user does not have a clearance for the ProgramFamily
+        with self.login(email=user.email):
+            response = self.client.get(url)
+            results = response.data["results"]
+            round_ids = [item["id"] for item in results]
+            self.assertFalse(the_round.id in round_ids)
+        # Give the user clearance for relevant ProgramFamily and check again
+        _add_clearance(user, the_round)
+        with self.login(email=user.email):
```

Comment:
I'd be more comfortable if these test cases were split into 2 separate tests eg `test_user_without_clearances_doesnt_see_judging_round_of_corresponding_program_families` and `test_user_with_clearances_sees_judging_rounds_of_corresponding_program_families`

Reply:
Even though the test case reads better with the "before and after", breaking it up as you suggest is the right thing to do. I'll do that.